### ISSUE TITLE
Auto / Manual Version Incrementing Fails when using Mobile Deployment from Git or FF

### ISSUE THREAD

**Issue by elevate-git** [FlutterFlow/flutterflow-issues#2340]
### Has your issue been reported?

- [X] I have searched the existing issues and confirm it has not been reported.
- [X] I give permission for members of the FlutterFlow team to access and test my project for the sole purpose of investigating this issue.

### Current Behavior

Versioning stopped working and will not build from GitHub since Friday 2/9. Error says Found build number 116 from TestFlight version 1.0.9 but pubspec.yaml says version: 1.0.9+1 which should create 1.0.9 build 17 automatically as per instructions.

I've tried the following: 

1) Unlink and link back Git 
2) Build from FF vs Git 
3) Tried adding App version and build version, also just one of them at a time or none. 
4) added the build version manually to pubspec.yaml
5) dart fix
6) Jumping versions to 1.9.10+1

### Expected Behavior

Should automatically increment build number using the format. When deploying from your GitHub branch, versioning is controlled by the version in your `pubspec.yaml` file. For example, to set the version number to 0.1.0 and build number to 2, use the following format: version: 0.1.0+2

### Steps to Reproduce

1) Build from Git branch main 

### Reproducible from Blank

- [X] The steps to reproduce above start from a blank project.

### Bug Report Code (Required)

NA

### Context

I have not been able to publish a new version since Friday. 

### Visual documentation

Codemagic Deploy Output
Failed Step: Flutter build ipa and automatic versioning

Found build number 116 from TestFlight version 1.0.9

Archiving com.elevatedaccess.app...
Upgrading contents.xcworkspacedata
Updating project for Xcode compatibility.
Upgrading project.pbxproj
Upgrading Runner.xcscheme
Upgrading Info.plist
Removing script build phase dependency analysis.
Adding input path to Thin Binary build phase.
Automatically signing iOS for device deployment using specified development team in Xcode project: 7FQDZN4R5X
Running pod install...                                             11.9s
Running Xcode build...                                          
Xcode archive done.                                         279.6s
Failed to build iOS app
Error (Xcode): lib/profile_page/profile_page_widget.dart:509:46: Error: The method 'where' isn't defined for the class 'int'.

Encountered error while archiving for device.

Build failed :|
Step 10 script `Flutter build ipa and automatic versioning` exited with status code 1

### Environment

```markdown
- FlutterFlow version: FlutterFlow v4.1 released February 9, 2024 Flutter version is 3.13.7
- Platform: Web (Mac)
- Browser name and version: Chrome Version 120.0.6099.234 (Official Build) (arm64)
- Operating system and version affected: iOS, Android
```

**Comment by hariprasadms** [comment]
HI @elevate-git - Thanks for submitting the issue. This issue is with dev team now and will be considered based on priority. Meanwhile you can try one workaround that, try changing the app version and using a new build number, example 2.0.1.  Let us know how it goes. Thanks

**Comment by hariprasadms** [comment]
Thanks for the update. The issue is already with the development team and will be addressed based on priority. We'll keep you posted on any updates. Thanks!

**Comment by JuanxCursed** [comment]
@elevate-git will gonna need to change directly on the xCode (iOS) or Vscode (Android) to make that work, from the FF platform, will not work

**Comment by elevate-git** [comment]
@JuanxCursed just clarify the version was updated in the pubspec.yaml file to version: 2.0.1+1 not inside FF. Only request to build from Git was done in FF.

**Comment by maxIrvine** [comment]
I'm also experiencing this issue. As are people in this forum: https://community.flutterflow.io/ask-the-community/post/step-10-script-flutter-build-ipa-and-automatic-versioning-exited-with-CCVNXghIMON4F3M?highlight=3txRbyKSRZozJe3

**Comment by elevate-git** [comment]
> I'm also experiencing this issue. As are people in this forum: https://community.flutterflow.io/ask-the-community/post/step-10-script-flutter-build-ipa-and-automatic-versioning-exited-with-CCVNXghIMON4F3M?highlight=3txRbyKSRZozJe3

@maxIrvine do you have any firebase queries where the 'Exclude Current User' switch is turned on? I was able to resolve the issue by turning off that switch and then successfully deployed. Confirmed that turning the switch back on produces the same versioning error which as you might have guessed has nothing to do with the firebase filter.

**Comment by maxIrvine** [comment]
@elevate-git what does the code look like for 'Exclude Current User'?

**Comment by elevate-git** [comment]
> @elevate-git what does the code look like for 'Exclude Current User'?

int columnCount = snapshot.data!.where((u) => u.uid != currentUserUid).toList(); vs int columnCount = snapshot.data!

**Comment by maxIrvine** [comment]
@elevate-git I searched my code base but couldn't find any code like that.

**Comment by MyTechAgent** [comment]
yep the same here. when you look at the build number when you manage to get a success you will notice it will be different again.

"we did try version jumping before and it did not work. I tried it again jumping to 2.0.1+1 and it still fails with the same error. Found build number 116 from TestFlight version 1.0.9"

**Comment by macxim1ananev** [comment]
Has anyone been able to solve this problem?

**Comment by MyTechAgent** [comment]
Yes,change version to something like 2.2.14 and version 140, when the status changes from building to publishing immediately change the version number only to something like 180.

**Comment by macxim1ananev** [comment]
@MyTechAgent 
![image](https://github.com/FlutterFlow/flutterflow-issues/assets/71488801/986f3104-18e5-42f9-bcd3-2536481f5479)
Here? could you describe the steps in more detail, I will be very grateful to you

**Comment by aburchfield** [comment]
Any update or workaround on this issue?

**Comment by JuanxCursed** [comment]
Still not working, any updates?

**Comment by ytfahmed** [comment]
> Any update or workaround on this issue?

Still not working, 
My workaround: 
- Turn off the deploy using Github
- Set the version and build manually
- Turn on deploy using Github again
![image](https://github.com/user-attachments/assets/03961772-2de5-4a19-aadb-1395e1dff28f)

**Comment by trevornocode** [comment]
> Still not working, My workaround:
> 
> * Turn off the deploy using Github
> * Set the version and build manually
> * Turn on deploy using Github again

This solved the issue! Thank you.

**Comment by JustUserHere** [comment]
@br0nk @takashiff 
any chance to resolve this bug please?
